backend/fms_core/services/sample.py

```diff
@@ -307,12 +307,30 @@ def extract_sample(process: Process,
 def prepare_library(process: Process,
                     sample_source: Sample,
                     container_destination: Container,
-                    library: Library,
+                    libraries_by_derived_sample,
                     volume_used,
                     execution_date: datetime.date,
                     coordinates_destination=None,
                     volume_destination=None,
                     comment=None):
+    """
+             Converts a sample into a library or a pool of samples into a pool of libraries.
+
+             Args:
+                 `process`: Process associated to the protocol.
+                 `sample_source`: The source sample to be converted.
+                 `container_destination`: The final volume of the sample (uL).
+                 `libraries_by_derived_sample`: A library object for each derived sample of the source sample.
```

Comment:
Describe the data structure... is it a dictionary ? what is the key ? The derived sample id ?